apiserver/objects.go

```diff
@@ -114,64 +111,116 @@ func (h *objectsCharmHTTPHandler) ServeGet(w http.ResponseWriter, r *http.Reques
 // Since juju's objects (S3) API only acts as a shim, this method will only
 // rewrite the http request for it to be correctly processed by the legacy
 // '/charms' handler.
-//
-// TODO(jack-w-shaw) Implement properly i.e. no longer shim around the legacy handler
 func (h *objectsCharmHTTPHandler) ServePut(w http.ResponseWriter, r *http.Request) error {
+	if r.Method != "PUT" {
```

Comment:
Not strictly. We should already know that have a PUT request, so this would only be caught by a programming error

However, it's not doing any harm and it keeps the handler consistent with others